eox_nelp/notifications/tests/test_notify_course_due_date.py

```diff
@@ -0,0 +1,138 @@
+        self,
+        course_enrollment_mock,
+        get_xblock_from_usage_key_string_mock,
+        get_xblock_course_from_usage_key_string_mock,
+        course_email_template_mock,
+        send_email_multialternative_mock,
+    ):
+        """Test correct behaviour of sending email with
+        method `notify_upcoming_course_due_date`.
+        Expected behavior:
+            - email_key_words are presented in the `text_message` arg (0) of 2nd call of course_email_template.
+            - email_key_words are presented in the `html_message` arg (0) of 3rd call of course_email_template.
+            - course_name is presented in `subject` arg(0) of the call of send_email_multialternative.
+            - subsection_name is presented in `subject` arg(0) of the call of send_email_multialternative.
+            - recipient list is presented in `recipient_emails` arg(3) of the call of send_email_multialternative.
+            - The email was sent from the correct email address (if specified)
+            - sent attribute of instance of `UpcomingCourseDueDate` after the process is True.
+            - Log successful of sent message.
+        """
+        UpcomingCourseDueDate.objects.all().delete()  # pylint: disable=no-member
+        instance, _ = UpcomingCourseDueDate.objects.get_or_create(  # pylint: disable=no-member
+            course=CourseOverview.objects.create(id="course-v1:test+Cx105+2022_T4"),
+            location_id="block-v1:test+CS501+2022_T4+type@sequential+block@a54730a9b89f420a8d0343dd581b447a",
+            due_date=self.subsection_due_date,
+            notification_date=timezone.now(),
+        )
+        sent_emails = 1
+        course_enrollment_mock.objects.filter.return_value = generate_list_mock_data(self.enrollments_data)
+        get_xblock_from_usage_key_string_mock.return_value = generate_list_mock_data(self.subsection_xblock_data)[0]
+        get_xblock_course_from_usage_key_string_mock.return_value = generate_list_mock_data(self.course_xblock_data)[0]
+        send_email_multialternative_mock.return_value = sent_emails
+        email_key_words = [
+            self.course_name,
+            self.subsection_name,
+            str(self.subsection_due_date),
+        ]
+
+        with self.assertLogs(notify_course_due_date.__name__, level="INFO") as logs:
+            notify_upcoming_course_due_date(instance)
+
+        for word in email_key_words:
+            self.assertIn(word, course_email_template_mock.mock_calls[1][1][0])
+            self.assertIn(word, course_email_template_mock.mock_calls[2][1][0])
+        self.assertIn(self.course_name, send_email_multialternative_mock.mock_calls[0][1][0])
+        self.assertIn(self.subsection_name, send_email_multialternative_mock.mock_calls[0][1][0])
+        self.assertListEqual(self.recipients_list, send_email_multialternative_mock.mock_calls[0][1][3])
+        self.assertTrue(instance.sent)
```

Comment:
you are not testing when `sent` is false

Reply:
Here ?
https://github.com/eduNEXT/eox-nelp/pull/44/files/fda7e2bc2e4ffc5055707b38b6c56845a3f515c2#diff-f2c383ced14e27737fb812392f5da8f4d4811351bc15cad4698a7052ac606953R134

Reply:
that tests when `get_xblock_course_from_usage_key_string` returns None, I refer when sent = send_email_multialternative(subject, plaintext_msg, html_msg, recipient_emails) returns None or False

Reply:
ok let me added a test to add hehe

Reply:
[19b22cb](https://github.com/eduNEXT/eox-nelp/pull/44/commits/19b22cb8d382753eebe58111a49bb76ed96a3b69)